code/game/objects/machinery/vending/vending.dm

```diff
@@ -386,7 +384,27 @@ GLOBAL_LIST_INIT(vending_white_items, typecacheof(list(
 
 	return TRUE
 
+/// Used only when vendor is tipped to put it back up
+/// Normal usage is in ui_interact
+/obj/machinery/vending/interact(mob/user)
+	. = ..()
+	if(.) // Handled by ui_interact
+		return
+	if(tipped_level == 2)
+		if(!iscarbon(user)) // AI can't heave remotely
+			return
+		user.visible_message("<span class='notice'> [user] begins to heave the vending machine back into place!</span>","<span class='notice'> You start heaving the vending machine back into place..</span>")
+		if(!do_after(user, 80, FALSE, src, BUSY_ICON_FRIENDLY))
+			return FALSE
+
+		user.visible_message("<span class='notice'> [user] rights the [src]!</span>","<span class='notice'> You right the [src]!</span>")
+		flip_back()
+		return TRUE
```

Comment:
```suggestion
	if(tipped_level != 2)
		return
	if(!iscarbon(user)) // AI can't heave remotely
		return
	user.visible_message("<span class='notice'> [user] begins to heave the vending machine back into place!</span>","<span class='notice'> You start heaving the vending machine back into place..</span>")
	if(!do_after(user, 80, FALSE, src, BUSY_ICON_FRIENDLY))
		return FALSE

	user.visible_message("<span class='notice'> [user] rights the [src]!</span>","<span class='notice'> You right the [src]!</span>")
	flip_back()
	return TRUE
```

Reply:
I will remove this check because it's already in `ui_interact` which has higher priority.